### PR TITLE
Feature/news reader

### PR DESCRIPTION
**Pull request by EdgarNX**

### CODE REVIEW

File: app/src/main/java/com/nemethedgar/newsreader/fragment/NewsListFragment.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    private AlertNavigator alertNavigator;
+
+    public static NewsListFragment newInstance() {
+        return new NewsListFragment();
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        //mViewModel = new ViewModelProvider(this).get(NewsReaderViewModel.class);
```
Comment (Misca) [inline app/src/main/java/com/nemethedgar/newsreader/fragment/NewsListFragment.java]: let's cleanup commented code

File: app/src/main/java/com/nemethedgar/newsreader/model/NewsReaderViewModel.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                .subscribe(
+                        this::onNewsArticlesReceived,
+                        this::onNewsArticlesError
+                );
+    }
+
+    private void onNewsArticlesReceived(@NonNull List<ArticleItemViewModel> articles) {
+        this.items.addAll(articles);
+//        isLoading.set(false);
+//        resultText.set(getApplication().getString(R.string.results, articles.size()));
```
Comment (Misca) [inline app/src/main/java/com/nemethedgar/newsreader/model/NewsReaderViewModel.java]: let's cleanup also here

File: data/src/main/java/com/nemethedgar/data/features/news/local/NewsEntity.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    @Nullable
+    public Integer id;
+
+    public String imageUrl;
+
+    public String title;
+
+    public String content;
+
+    public String description;
```
Comment (Misca) [inline data/src/main/java/com/nemethedgar/data/features/news/local/NewsEntity.java]: let's make all the members final, besides the id: immutable classes are less prone to bugs, considering their state does not change trough their lifecycle

File: data/src/main/java/com/nemethedgar/data/features/news/local/NewsLocalDataStore.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Single;
+import io.reactivex.schedulers.Schedulers;
+
+public class NewsLocalDataStore {
+
+    private static NewsDao dao;
```
Comment (Misca) [inline data/src/main/java/com/nemethedgar/data/features/news/local/NewsLocalDataStore.java]: let's make the dao final but static, why is it static? We should avoid static members, as are less testable, prone to memory leaks

**Review by Misca — CHANGES_REQUESTED** [review]
allright, a few changes away 👍

File: app/src/main/java/com/nemethedgar/newsreader/fragment/NewsListFragment.java (hunk 1 above)
Comment (EdgarNX) [reply app/src/main/java/com/nemethedgar/newsreader/fragment/NewsListFragment.java]: done

File: app/src/main/java/com/nemethedgar/newsreader/model/NewsReaderViewModel.java (hunk 2 above)
Comment (EdgarNX) [reply app/src/main/java/com/nemethedgar/newsreader/model/NewsReaderViewModel.java]: done

File: data/src/main/java/com/nemethedgar/data/features/news/local/NewsEntity.java (hunk 3 above)
Comment (EdgarNX) [reply data/src/main/java/com/nemethedgar/data/features/news/local/NewsEntity.java]: done

File: data/src/main/java/com/nemethedgar/data/features/news/local/NewsLocalDataStore.java (hunk 4 above)
Comment (EdgarNX) [reply data/src/main/java/com/nemethedgar/data/features/news/local/NewsLocalDataStore.java]: done

**Review by Misca — APPROVED** [review]
well done 👍

**Pull request merged**